app/models/manageiq/providers/cloud_manager/provision_workflow.rb

```diff
@@ -31,6 +23,12 @@ def allowed_cloud_subnets(_options = {})
     end
   end
 
+   def allowed_cloud_networks(_options = {})
+     source = load_ar_obj(get_source_vm)
+     targets = get_targets_for_source(source, :cloud_filter, CloudNetwork, 'cloud_network_id')
```

Comment:
'cloud_network_id' should be changed to the association needed to return the targets.  e.g. source.cloud_networks - so 'cloud_networks'. The method sends that string off the `source`.